scanner.h

```diff
@@ -0,0 +1,119 @@
+//
+// Created by loco on 22/10/2020.
+//
+#pragma once
+#include <string>
+#include <regex>
+#include <iostream>
+using namespace std;
+vector<string> separators = {",", ";", "(", ")", "{", "}", "[", "]"};
+vector<string> operators = {"<=", ">=", "<", ">", "=", "+", "-", "/", "*", "%"};
+vector<string> reservedWords = {"INTREG",
+                                "CARACTER",
+                                "INTERVAL",
+                                "ADEVAR",
+                                "ADEVARAT",
+                                "MINCIUNA",
+                                "BAGA",
+                                "READ_FROM_USER",
+                                "PRINT_TO_USER",
+                                "DECLAR",
+                                "DACA",
+                                "ATUNCI",
+                                "INCEPE_SMECHERIA",
+                                "GATA_SMECHERIA",
+                                "TARE",
+                                "DEFINITIE"};
+enum Type{
+    resW, oper, sep
+};
+bool checkStrings(const string &token, Type t)  {
+    bool isOkay = false;
+    vector<string> givenFilter;
+    switch (t) {
+        case 0: givenFilter = reservedWords; break;
+        case 1: givenFilter = operators; break;
+        case 2: givenFilter = separators; break;
+    }
+    for(auto s : givenFilter) {
+        if(token.compare(s) == 0){
+            isOkay = true;
+        };
+    }
+    return isOkay;
+}
+
+bool isConstant(string token) {
+    std::regex cifre("^[0-9]*$", std::regex_constants::ECMAScript | std::regex_constants::icase);
+    std::regex caractereString("^(?:\".*\"$)$", std::regex_constants::ECMAScript | std::regex_constants::icase);
```

Comment:
Here you can add some comments just to be sure that in the future you will understand what you want to do here